mod/discussions/start.php

```diff
@@ -647,7 +647,8 @@ function discussion_topic_container_permissions($hook, $type, $value, $params) {
 	$subtype = elgg_extract('subtype', $params);
 	$container = elgg_extract('container', $params);
 
-	if (!$value || !$container || $subtype !== 'discussion') {
+	if (!$value || !$container || $subtype !== 'discussion' ||
+			$container == elgg_get_logged_in_user_entity()) {
```

Comment:
This will allow users to create discussions even if the "user" type isn't approved.